### ISSUE TITLE
[engine] cant build for x86 Android emulator from m1 mac.

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#133059]
Trying to build for an x86 emulator build from an m1 mac fails due to a compilation error in third_party/dart.

Specifically for the following gn config:

`./tools/gn --unopt --android-cpu=x86 --android`

This will eventually fail with:

```
In file included from ../../third_party/dart/runtime/bin/eventhandler.cc:5:
In file included from ../../third_party/dart/runtime/bin/eventhandler.h:8:
In file included from ../../third_party/dart/runtime/bin/builtin.h:13:
In file included from ../../third_party/dart/runtime/platform/assert.h:8:
../../third_party/dart/runtime/platform/globals.h:348:2: error: Mismatched Host/Target architectures.
#error Mismatched Host/Target architectures.
 ^
1 error generated.

```

Where the line in question is:

```
// Verify that host and target architectures match, we cannot
// have a 64 bit Dart VM generating 32 bit code or vice-versa.
#if defined(TARGET_ARCH_X64) || defined(TARGET_ARCH_ARM64) ||                  \
    defined(TARGET_ARCH_RISCV64)
#if !defined(ARCH_IS_64_BIT) && !defined(FFI_UNIT_TESTS)
#error Mismatched Host/Target architectures.
```

So it seems like there must be multiple architecture defines enabled (arm64 from m1 and x86 from build config). So perhaps there are some baked in assumptions in the dart config?

**Comment by zanderso** [comment]
@jonahwilliams Could you explain a bit why you need this?

**Comment by jonahwilliams** [comment]
@reidbaker was attempting to build an x86 local engine to test a change that we only have emulator coverage for, IIRC for older API levels we rely on emulation.

**Comment by reidbaker** [comment]
Yes if it helps prioritize or give some context specifically I was trying to test a local engine change against an x86 emulator of android compiling on an m1 Mac.

**Comment by zanderso** [comment]
Sorry if I'm missing something, but why does it need to be an x86 emulator to test against an older API level? Could it be a 32-bit arm emulator?

**Comment by reidbaker** [comment]
That is what the defaults were. I was able to make an arm 64 bit emulator
and ran into a different issue. If building for x86 on an arm mac is not
something we support then saying that would be better than the error I got.
